### ISSUE TITLE
Boost exception while loading files

### ISSUE THREAD

**Issue by wgrriffel** [otland/forgottenserver#2018]
### Before creating an issue, please ensure:
- [x] This is a bug in the software that resides in this repository, and not a
      support matter (use https://otland.net/forums/support.16/ for support)
- [x] This issue is reproducible without changes to the code in this repository

### Steps to reproduce (include any configuration/script required to reproduce)
1. Game.loadMap("") to a nonexistent path or rename map/items otb
2. Crash

It will crash if path or file doest exist.

### Expected behaviour
Should display error output instead and not crash

### Actual behaviour
`theforgottenserver.exe!boost::throw_exception<std::ios_base::failure>(const std::ios_base::failure & e) Line 69`

### Environment
Win7/Debian 8

**Comment by djarek** [comment]
https://github.com/otland/forgottenserver/blob/master/src/iomap.cpp#L72
I think the construction of the loader should be wrapped with a try ... catch block, an error should be printed, and the exception should probably be rethrown. Could you confirm the same happens if the item.otb file is missing?

**Comment by wgrriffel** [comment]
@djarek yes it does the same, exeption thrown at `theforgottenserver.exe!boost::throw_exception<std::ios_base::failure>(const std::ios_base::failure & e) Line 69`

**Comment by wgrriffel** [comment]
@djarek even so, don't you think it's an overkill terminate the server for this reason? i mean for files that are not a dependency of server, such as loading a new map with Game.loadMap.

**Comment by wgrriffel** [comment]
@dalkon btw don't you think it's an overkill throw an exeption because of a bad file path? i'm not about core files loaded on server startup, im about simply map loaded in Game.loadMap

**Comment by djarek** [comment]
I guess it's safe to rework the error handling in the OTB parser, because failure in OTB tree construction does not alter the map data structures.
